docs/cross-account-resource-management.md

```diff
@@ -0,0 +1,56 @@
+# Cross account resource management (CARM)
+
+## The problem
+
+Today adminstrative users can manage AWS resources using ACK. However, these resources are created in the same AWS account as the Kubernetes cluster. This leads to a suboptimal experience for users in large organisations: creating resources in multiple AWS accounts requires ACK to be installed and managed in many kubernetes clusters. Considering ACK has dozens of service controllers, this means administrators must manage hundreds of AWS service controller pods across many kubernetes clusters and many AWS Accounts with scoped roles.
+
+With cross account resource management, a single kubernetes cluster with ACK installed can manage the resources of many different AWS accounts. Kubernetes administrators will not need to install and manage multiple clusters and ACK controllers, and AWS administrators will spend less effort managing permissions to access the services across accounts.
+
+## Proposed implementation
+
+The proposed solution will solve the problem of cross account management within a single kubernetes cluster. Given an AWS Owner account ID it will look up an IAM Role ARN and call STS:AssumeRole to pivot the AWS SDK calls in order to execute in a new AWS Account and role context.
+
+Controllers will store AccountIDs and their associated IAM Role ARNs in a ConfigMap or Secret Kubernetes object. Cluster administrators will be responsible for administrating and updating the configmap/secret, and creating the target roles using the AWS Console/CLI. Controllers can reuse and existing IAM Role ARN if it matches the scope they need. 
+
+![carm-flowchart](./images/carm-flowchart.png)
+
+## Potential other solutions
+
+- Store the IAM Role ARNs and account IDs as a kubernetes objects (new CRDs)
```

Comment:
IAM Role ARN format already has account ID, region information. Does user/admin required to pass Account ID(s) explicitly ? and why ?

Reply:
This is more for simplifying user/admin experience with ACK, they will only have to provide an Account ID. We can also support both methods. WDYT ?

Reply:
Make sense! Thanks for clarifying.